Clean up Verilator driver/tasks (SC-345, SC-346)

nmoroze opened the pull request:
- Remove parse task: redundant with Surelog and is unused
- Allow both compile and lint to read input from flow or parameters
- Update docs
- Update lintflow to run Verilator directly on input files
- Add test of lintflow
- Delete redundant Verilator test

gadfort commented on siliconcompiler/tools/verilator/compile.py:
```diff
@@ -25,12 +25,22 @@ def setup(chip):
+
+    cmdlist = runtime_options_tool(chip)
+
+    for value in chip.find_files('input', 'hll', 'c', step=step, index=index):
+        cmdlist.append(value)
```
should you add input,hll,c as a required field for this?

nmoroze replied:
good catch - it's not required to provide a C/C++ file to Verilator in this mode but I'll add the "if chip.valid(...) then require" pattern since that'll check that the filepath is valid if it's provided.

gadfort commented on tests/flows/test_lintflow.py:
```diff
@@ -0,0 +1,39 @@
+    chip.run()
+
+
+@pytest.mark.quick
+@pytest.mark.eda
+def test_lint_post_surelog(scroot):
+    chip = siliconcompiler.Chip('heartbeat')
+
+    v_src = os.path.join(scroot, 'tests', 'data', 'heartbeat.v')
+    chip.input(v_src)
+
+    flow = 'lint'
+    chip.node(flow, 'import', parse)
+    chip.node(flow, 'lint', lint)
+    chip.edge(flow, 'import', 'lint')
+    chip.set('option', 'flow', flow)
+
+    chip.run()
```
this test is specific for verilator, so we should pull it out into the tools tests directory

gadfort reviewed (changes requested):
please move the tests into their appropriate categories.

nmoroze commented on siliconcompiler/tools/verilator/lint.py:
```diff
@@ -16,9 +14,6 @@ def setup(chip):
-    chip.add('tool', tool, 'task', task, 'option', ['--lint-only', '--debug'],
-             step=step, index=index)
-    chip.set('tool', tool, 'task', task, 'input', f'{design}.v',
+    chip.add('tool', tool, 'task', task, 'option', ['--lint-only'],
```
seems like core.py will fall back to tool driver `runtime_options` if unspecified. If you plan to change that behavior and/or you just think this should be more consistent with compile.py I'll fix

gadfort replied:
the fall back is not guaranteed to stay around

nmoroze commented:
Addressed feedback! I also snuck in another feature (support for Verilator config files via [tool, task, file]... these are helpful for waiving lint violations in 3rd party code)

gadfort reviewed (approved):
Lgtm
